### DISCUSSION TITLE
Collect ideas for improving user experience

### DISCUSSION THREAD

**Discussion by dominikriemer** [apache/streampipes#828] [Ideas]
Let's see how we can improve user experience!

What are the biggest pain points regarding usability?
Where do you see improvement potentials?
Where do you get errors that are hard to understand and where are error messages missing?

**Comment by dominikriemer** [comment]
One thing we could improve is the interactive tutorial, which only opens in the pipeline editor and requires the watertank simulator service we don't ship by default anymore.
Instead, the interactive tutorail should open on the main screen the first time a user logs in, guide the user to StreamPipes connect to create a new data source, and then guide the users to the data explorer, pipeline editor and other important views.

**Reply by tenthe** [reply]
Hello Dominik,
I think this is a good question. I was wondering the same thing.

Should we have one big tutorial that covers everything (create an adapter, build a pipeline and display the data in the dashboard/data explorer) or should we have a single tutorial for each module. If you have multiple tutorials, the order in which the tutorials must be performed may be predetermined.

Currently there is a small icon in each module that launches a tutorial, but I'm not sure if users can see this, so a pop up might be helpful.

**Comment by flomickl** [comment]
I have two points.
1) In the pipeline editor:

sometimes the hint texts are 1, 2 or sometimes 3 columns high; during scrolling this is a little bit irritating for me and during scrolling i don't need this information. I can group the elements in categories, but I can't switch them of or make them for a short moment invisible. So I have a better overview and can reduce the big amount of options (without installing uninstalling them).

![animation_in_pipeline_editor](https://user-images.githubusercontent.com/16739072/205508112-b234233c-6856-460c-bef3-304b5e031e75.gif)

2) During Installation. I can filter for processors but not for groups of processors. Also I can't multi select elements and install them. I only have the options select all + in/ uninstall selected. So I have to install them step by step and after each installation the side is refreshing and I have to scroll from the beginning again.

![instalaltions_without_category](https://user-images.githubusercontent.com/16739072/205508254-13a3f0e7-9775-4739-ab26-4679fb7b7cd8.gif)

Greetings
Florian

**Comment by flomickl** [comment]
I found another UI point 
If I want to add a semantic type in the adapters, the field is too small to see the full length.
Either to see the full length or just the last bit of it with the important vocabulary.

![Auswahl_040](https://user-images.githubusercontent.com/16739072/211374958-a14195d9-dc5f-4b77-a17e-64d544d60c3c.png)

**Comment by dominikriemer** [comment]
Thanks @flomickl!
Can you create a ticket for item 3? This should be rather easy to fix and we can mark it as good first issue. 
Conerning point 2, you can select multiple elements in the installation view, just click on each item and then click "install selected".
For your first point we need to discuss as I think there are several options to do that (we had a concept of "Favorites" some time ago where you could pin your favorite elements, which could be interesting here, but also improved search would be an option).

**Reply by flomickl** [reply]
It was hard to find this post again ^^
I created an issue as mentioned for item 3 ->  #1199 

Item 2 : Yes, you can select multiple elements, but it is not really highlighted and visible which items are selected from the user. A little bigger frame around the selection would help. What do you think?

Item 1: Yes, this is really a complex thing to do plan and do. Maybe I'll have a chat with a UI specialist friend of mine. Maybe he has some ideas.
